### PR TITLE
17831 - EFT Monthly Statement

### PR DESCRIPTION
**Pull request by ochiu**
*Issue #:*
https://github.com/bcgov/entity/issues/17831

*Description of changes:*
- Update logic to publish new statement notification for EFT
- Add feature flag logic to enable EFT processing
- Add basic feature flag tests
- Add statement notification tests for previous and new logic 
- Add publish to mailer helper (previous template is within payment jobs)

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the sbc-pay license (Apache 2.0).

### CODE REVIEW

**Comment by ochiu** [comment]
Fixing tests...

File: jobs/payment-jobs/tasks/statement_notification_task.py [hunk 1]
```diff
@@ -78,14 +80,27 @@ def send_notifications(cls):
             params['frequency'] = statement.frequency.lower()
             # logic changed https://github.com/bcgov/entity/issues/4809
             # params.update({'url': params['url'].replace('orgId', payment_account.auth_account_id)})
+
+            notification_success = True
+            eft_enabled = flags.is_on('enable-eft-payment-method')
```
Comment (seeker25) [inline jobs/payment-jobs/tasks/statement_notification_task.py:85]: I'd recommend using the default `flags.is_on('enable-eft-payment-method', False)`

Comment (ochiu) [reply jobs/payment-jobs/tasks/statement_notification_task.py]: makes sense, updated to specify the default explicitly

Comment (seeker25) [reply jobs/payment-jobs/tasks/statement_notification_task.py]: thank you

image

File: jobs/payment-jobs/utils/mailer.py [hunk 2]
```diff
@@ -55,3 +56,40 @@ def publish_mailer_events(message_type: str, pay_account: PaymentAccountModel,
...
+    payload = {
+        'specversion': '1.x-wip',
+        'type': f'bc.registry.payment.statementNotification',
+        'source': f'https://api.pay.bcregistry.gov.bc.ca/v1/accounts/{pay_account.auth_account_id}',
+        'id': f'{pay_account.auth_account_id}',
+        'time': f'{datetime.now()}',
+        'datacontenttype': 'application/json',
+        'data': {
+            'emailAddresses': emails,
+            'accountId': pay_account.auth_account_id,
```
Comment (seeker25) [inline jobs/payment-jobs/utils/mailer.py:73]: auth_account_id is a string at the model level I think - so no str() needed

File: jobs/payment-jobs/tests/services/test_flags.py [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+"""Test-Suite to ensure that the Flag Service is working as expected."""
+import pytest
+from flask import Flask
+
+from pay_api.services import Flags
+
+app = None
+
+
+@pytest.fixture
```
Comment (seeker25) [inline jobs/payment-jobs/tests/services/test_flags.py:24]: Do we need these tests again? they're in pay already

**Review by seeker25 — APPROVED** [review]
Up the version number if possible?

**Pull request merged**